staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/equality.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fieldmanager
+
+import (
+	"context"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var equalities = func() conversion.Equalities {
+	e := equality.Semantic
+	e.AddFunc(ignoreManagedFieldsTimestamp)
```

Comment:
this changes a process-wide equality 😱

Reply:
at least it touches the array below the slice, nearly as bad. Please do a deep copy of the slice first.

Reply:
Background:
```
e1 := equality.Semantic
e1.AddFunc(ignoreManagedFieldsTimestamp)
e2 := equality.Semantic
e2.AddFunc(someOtherEqualityFunc)
```
Which equality will be used in e1 and e2?

Reply:
Nice catch, the map is now being shallow copied